Zend/zend_closures.c

```diff
@@ -74,6 +74,12 @@ static bool zend_valid_closure_binding(
 {
 	zend_function *func = &closure->func;
 	bool is_fake_closure = (func->common.fn_flags & ZEND_ACC_FAKE_CLOSURE) != 0;
+	
+	if (func->common.fn_flags & ZEND_ACC_ABSTRACT) {
```

Comment:
My baseline expectation here would have been that partials are treated as "fake closures" (what `Closure::fromCallable()` returns) and be subject to the same limitations. Which would mean, in particular, that while most rebinding is forbidden, you *can* rebind to a compatible object (different object of same class, or an object of a sub-class). Any particular reason to deviate from that?

Reply:
I stopped it there originally because we'll end up in a function that can't create a zend_partial, or copy it, we needed an API, also $this was inconsistent in the version you commented on.

I've done that preliminary work, we now have an API that could be used, and $this is consistently correct.

Still it's not done yet, but we're totally able to do it now ... I'll get there ...